pkg/sql/discard.go

```diff
@@ -84,20 +86,23 @@ func (n *discardNode) startExec(params runParams) error {
 }
 
 func deleteTempTables(ctx context.Context, p *planner) error {
-	codec := p.execCfg.Codec
-	descCol := p.Descriptors()
-	allDbDescs, err := descCol.GetAllDatabaseDescriptors(ctx, p.Txn())
-	if err != nil {
-		return err
-	}
-	ie := p.execCfg.InternalExecutor
-
-	for _, dbDesc := range allDbDescs {
-		schemaName := p.TemporarySchemaName()
-		err = cleanupSchemaObjects(ctx, p.execCfg.Settings, p.Txn(), descCol, codec, ie, dbDesc, schemaName)
+	if err := p.WithInternalExecutor(ctx, func(ctx context.Context, txn *kv.Txn, ie sqlutil.InternalExecutor) error {
```

Comment:
nit: could just be `return p.WithInternalExecutor(...`

Reply:
Done.